data-avro/src/main/java/com/linkedin/data/avro/SchemaTranslator.java

```diff
@@ -166,9 +171,12 @@ public static DataSchema avroToDataSchema(String avroSchemaInJson, AvroToDataSch
             {
               avroSchemaFromEmbedded.addProp(DATA_PROPERTY, embededSchemaPropertyVal);
             }
-            if (!avroSchemaFromEmbedded.equals(avroSchemaFromJson))
-            {
-              throw new IllegalArgumentException("Embedded schema does not translate to input Avro schema: " + avroSchemaInJson);
+            // Compare using configuration equivalent to STRICT, except ignore TRANSLATED_FROM_SOURCE_OPTION
+            if (!ConfigurableSchemaComparator.equals(avroSchemaFromEmbedded, avroSchemaFromJson,
+                new SchemaComparisonConfiguration(true, true, true, false, true, true,
+                    Collections.singleton((TRANSLATED_FROM_SOURCE_OPTION))))) {
```

Comment:
This is a bit hard to read. Why can you provide a constructor in `ConfigurableSchemaComparator` to allow configure what options to be excluded? Also why `DATA_PROPERTY` above is not handled simiarly?

Reply:
This check will never fail for `DATA_PROPERTY` because the code above this check adds it just before the check.
```java
Object embededSchemaPropertyVal = avroSchemaFromJson.getObjectProp(DATA_PROPERTY);
            if (embededSchemaPropertyVal != null)
            {
              avroSchemaFromEmbedded.addProp(DATA_PROPERTY, embededSchemaPropertyVal);
            }
```

Updated the SchemaComparisonConfiguration